docs/Design/For Each.md

```diff
@@ -0,0 +1,155 @@
+# For Each
+A loop that iterates over all elements of an array. This loop does not use an explicit index.
+
+## Commands
+
+### `for each start`
+
+`for each start : array type iterator`
+
+This command will be used to start a for each code block. The first parameter is the array to be iterated over.
+The second command is the type of the array. The third parameter is the iterator variable.
+
+### `for each end`
+
+This command ends a for each code block.
+
+## Usage
+
+```
+for each start : arrayvar int item
```

Comment:
`arrayvar` is an abbreviated name, which is against GLS' coding conventions. Call it `myArray` instead.